hooks/post-command

```diff
@@ -25,6 +25,135 @@ plugin_read_list_into_result() {
+      return
+    elif [ -f /etc/redhat-release ] ; then
+      DIST='RedHat'
+      PSUEDONAME=$(sed s/.*\(// < /etc/redhat-release | sed s/\)//)
+      REV=$(sed s/.*release\ // < /etc/redhat-release | sed s/\ .*//)
+      OSSTR="${OS} ${DIST} ${REV}(${PSUEDONAME} ${KERNEL} ${MACH})"
+      echo ${OSSTR}
+      return
+    elif [ -f /etc/SuSE-release ] ; then
+      DIST=$(tr "\n" ' ' < /etc/SuSE-release | sed s/VERSION.*//)
+      REV=$(tr "\n" ' ' < /etc/SuSE-release| sed s/.*=\ //)
+      OSSTR="${OS} ${DIST} ${REV}(${KERNEL} ${MACH})"
+      echo ${OSSTR}
+      return
+    elif [ -f /etc/mandrake-release ] ; then
+      DIST='Mandrake'
+      PSUEDONAME=$(sed s/.*\(// < /etc/mandrake-release | sed s/\)//)
+      REV=$(sed s/.*release\ // < /etc/mandrake-release | sed s/\ .*//)
+      OSSTR="${OS} ${DIST} ${REV}(${KERNEL} ${MACH})"
+      echo ${OSSTR}
+      return
+    elif [ -f /etc/debian_version ] ; then
+      if [ "$(awk -F= '/DISTRIB_ID/ {print $2}' /etc/lsb-release)" = "Ubuntu" ]; then
+        DIST="Ubuntu"
+      else
+        DIST="Debian $(cat /etc/debian_version)"
+      fi
+      OSSTR="${OS} ${DIST} (${KERNEL} ${MACH})"
+      echo ${OSSTR}
+      return
+    else
+      echo $(uname -a)
+      return
+    fi
+    if [ -f /etc/UnitedLinux-release ] ; then
+      DIST="${DIST}[$(tr "\n" ' ' < /etc/UnitedLinux-release | sed s/VERSION.*//)]"
+      OSSTR="${OS} ${DIST}(${KERNEL} ${MACH})"
+      echo ${OSSTR}
+      return
+    fi
+
+  elif [ "${OS}" == "Darwin" ]; then
+    type -p sw_vers &>/dev/null
+    [ $? -eq 0 ] && {
+      OS=`sw_vers | grep 'ProductName' | cut -f 2`
+      VER=`sw_vers | grep 'ProductVersion' | cut -f 2`
+      BUILD=`sw_vers | grep 'BuildVersion' | cut -f 2`
+      OSSTR="Darwin ${OS} ${VER} ${BUILD}"
+    } || {
+      OSSTR="MacOSX"
+    }
+    echo ${OSSTR}
+    return
+  fi
+
+  echo $(uname -a)
+}
+
+get_codecov_uploader() {
+  local os=$(get_os)
+  local os_array=($os)
+  if [ -n ${os_array[1]} ] && [ ${os_array[1]} = "Alpine" ]; then
+    apk add gnupg
+    curl https://keybase.io/codecovsecurity/pgp_keys.asc | gpg --no-default-keyring --keyring trustedkeys.gpg --import # One-time step
+    curl -Os https://uploader.codecov.io/latest/alpine/codecov
+    curl -Os https://uploader.codecov.io/latest/alpine/codecov.SHA256SUM
+    curl -Os https://uploader.codecov.io/latest/alpine/codecov.SHA256SUM.sig
+    gpgv codecov.SHA256SUM.sig codecov.SHA256SUM
+    sha256sum -c codecov.SHA256SUM
+    chmod +x codecov
+    return
+  elif [ -n ${os_array[0]} ] && [ ${os_array[0]} = "Linux" ]; then
+    curl https://keybase.io/codecovsecurity/pgp_keys.asc | gpg --no-default-keyring --keyring trustedkeys.gpg --import # One-time step
+    curl -Os https://uploader.codecov.io/latest/linux/codecov
+    curl -Os https://uploader.codecov.io/latest/linux/codecov.SHA256SUM
+    curl -Os https://uploader.codecov.io/latest/linux/codecov.SHA256SUM.sig
+    gpgv codecov.SHA256SUM.sig codecov.SHA256SUM
```

Comment:
This makes gpg a prereq for the plugin. We should name it in the plugin.yml

Reply:
Fixed in https://github.com/joscha/codecov-buildkite-plugin/pull/19/commits/388c47dc57d487f625544c2c9532c6b66cc2b2e2